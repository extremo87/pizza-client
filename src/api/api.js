import axios from 'axios';


const ServerResponseCode = {
  UNAUTHORIZED: 401,
  VALIDATION: 422
};

export const createAPI = (onError) => {
  const api = axios.create({
    baseURL: process.env.API_URL,
    timeout: 5000
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (error) => {
    if (error.response) {
      if (error.response.status === ServerResponseCode.VALIDATION) {
        const errors = [];

        Object.keys(error.response.data.errors).map((key) => {
          errors.push(error.response.data.errors[key]);
        });

        onError(errors.join(`,`));

      } else {
        onError(error.message);
      }
    } else {
      onError(error.message);
    }
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
