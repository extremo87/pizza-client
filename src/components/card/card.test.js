import React from 'react';
import Card from './card';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NameSpace from "../../reducers/name-space.js";
import thunk from 'redux-thunk';
import {createAPI} from "../../api/api";
import {BrowserRouter} from 'react-router-dom';


const api = createAPI(() => {});

const mockStore = configureStore([thunk.withExtraArgument(api)]);

const product = {
  id: 1,
  title: `MARGHERITA`,
  ingredients: `Tomatoes, mozzarella, ricotta, bacon, onion`,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
  image: `https://static.nnc.kz/pizza1.jpg`,
  price: 18,
  created_at: `2020-09-21 13:18:35`,
  updated_at: `2020-09-21 13:18:35`
};


const store = mockStore({
  [NameSpace.CART]: {
    cart: {},
  },
  [NameSpace.DATA]: {
    products: [],
  }
});


it(`Product card conponent renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Card
              product={product}
              currencySymbol={`$`}
              key={product.id}
            />
          </BrowserRouter>
        </Provider>

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
