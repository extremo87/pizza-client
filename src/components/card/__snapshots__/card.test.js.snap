// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card conponent renders correctly 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card mb-4 shadow-sm"
  >
    <img
      className="card-img-top"
      src="https://static.nnc.kz/pizza1.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        <a
          className="text-success"
          href="/product/1"
          onClick={[Function]}
        >
          MARGHERITA
        </a>
      </h5>
      <p
        className="card-text"
      >
        Tomatoes, mozzarella, ricotta, bacon, onion
      </p>
      <div
        className="d-flex justify-content-between align-items-center"
      >
        <div
          className="btn-group"
        >
          <button
            className="btn btn-sm btn-outline-secondary add-to-cart"
            onClick={[Function]}
            type="button"
          >
            Add to cart
          </button>
        </div>
        <p>
          <b>
            $
            18
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;
